lib/src/rules/always_require_non_null_named_parameters.dart

```diff
@@ -106,18 +124,23 @@ class Visitor extends SimpleAstVisitor {
     }
   }
 
-  bool _hasAssertNotNull(AssertStatement node, String name) {
+  bool _hasAssertNotNull(Expression node, String name) {
     bool _hasSameName(Expression rawExpression) {
       final expression = rawExpression.unParenthesized;
       return expression is SimpleIdentifier && expression.name == name;
     }
 
-    final expression = node.condition.unParenthesized;
-    if (expression is BinaryExpression &&
-        expression.operator.type == TokenType.BANG_EQ) {
-      final operands = [expression.leftOperand, expression.rightOperand];
-      return operands.any(DartTypeUtilities.isNullLiteral) &&
-          operands.any(_hasSameName);
+    final expression = node.unParenthesized;
+    if (expression is BinaryExpression) {
+      if (expression.operator.type == TokenType.AMPERSAND_AMPERSAND) {
```

Comment:
No, because `assert(param1 != null || param2 != null)` doesn't mean that both `param1` and `param2` have to be required.